File: perl/10.pl
#!/usr/bin/perl
use List::Util qw(sum);
use strict;

# Find the sum of all the primes below two million.

sub sieve_of_eratosthenes{
  my $N = shift;
  my @list = 0..$N;

  for(my $i = 2; $i < $N; $i++){
    if($list[$i]){
      for(my $j = $i; $j * $i <= $N; $j++){
        $list[$i * $j] = 0;
      }
    }
  }
  return grep { $_ > 1 } @list;
}


print sum sieve_of_eratosthenes(2000000);
print "\n";

File: javascript/10.js
#!/usr/bin/env node
var _ = require('./underscore');

// Find the sum of all the primes below two million.

function sieve_of_eratosthenes(n){
  var list = _.range(n + 1);

  for(var i = 2; i < n; i++){
    if(list[i]){
      for(var j = i; j * i <= n; j++){
        list[i * j] = 0;
      }
    }
  }
  return _.filter(list, function (e){ return e > 1 });
}

var primes = sieve_of_eratosthenes(2000000);
console.log( _.reduce(primes, function (a, b){ return a + b }) );




